chef/automate
# CHEF-721: Preflight-check from bastion

### :nut_and_bolt: Description: What code changed, and why?

Made changes to support chef-automate preflight-check command from bastion. The following commands are supported from bastion :
```
chef-automate preflight-check --ha-deployment-config <path to your config.toml on bastion> --automate
chef-automate preflight-check --ha-deployment-config <path to your config.toml on bastion> --a2
chef-automate preflight-check --ha-deployment-config <path to your config.toml on bastion> -a
chef-automate preflight-check --ha-deployment-config <path to your config.toml on bastion> --chef_server
chef-automate preflight-check --ha-deployment-config <path to your config.toml on bastion> --cs
chef-automate preflight-check --ha-deployment-config <path to your config.toml on bastion> -c
chef-automate preflight-check --ha-deployment-config <path to your config.toml on bastion> --fe
chef-automate preflight-check --ha-deployment-config <path to your config.toml on bastion> --frontend
chef-automate preflight-check --ha-deployment-config <path to your config.toml on bastion> --chef_server --node 
<ip address of node>
chef-automate preflight-check --ha-deployment-config <path to your config.toml on bastion> --automate --node 
<ip address of node>
chef-automate preflight-check --ha-deployment-config <path to your config.toml on bastion> --fe --node 
<ip address of node>
chef-automate preflight-check --ha-deployment-config <path to your config.toml on bastion> --automate --airgap
chef-automate preflight-check --ha-deployment-config <path to your config.toml on bastion> --chef_server --airgap
chef-automate preflight-check --ha-deployment-config <path to your config.toml on bastion> --fe --airgap

```
### :chains: Related Resources

https://chefio.atlassian.net/browse/CHEF-721

### :+1: Definition of Done
`chef-automate preflight-check` is available on bastion 

### :athletic_shoe: How to Build and Test the Change

1. Checkout this branch
2. rebuild components/automate-cli
3. upload the build hab pkg
4. install this hab pkg in an HA cluster
5. execute the preflight-check commands from bastion

### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
[Preflight-check video](https://progresssoftware.sharepoint.com/:v:/s/ChefCoreC/ESsUtdmYj19CnSyS8iQnfIYBgpDTdeJ7o5X1OQZHGOIdEg?e=hZOyag)
